sensor/tests/resource/helper.go

```diff
@@ -338,6 +370,24 @@ func GetAllAlertsForDeploymentName(messages []*central.MsgFromSensor, name strin
 	return selected
 }
 
+// GetAllAlertsForDeploymentID filters sensor messages and gets all alerts for a deployment with `id`. If checkEmptyAlerts is set to true it will also check for AlertResults with no Alerts
+func GetAllAlertsForDeploymentID(messages []*central.MsgFromSensor, id string, checkEmptyAlertResults bool) []*central.MsgFromSensor {
```

Comment:
I think this logic can be simplified. If you're only interested in the last alert, you don't need to have a function that returns _all_ alerts for a deployment and then pick the last element on the list. The pre-filter is unnecessary.

My suggestion is to simply do the following:
```go
func GetLastViolationStateForDeploymentID(messages []*central.MsgFromSensor, id string, func GetAllAlertsForDeploymentID(messages []*central.MsgFromSensor, id string, checkEmptyAlertResults bool) []*central.MsgFromSensor {) *central.MsgFromSensor {
        for i := len(messages) - 1; i >= 0; i-- {
                // check empty and check ID, if all good return here.
        }
}
```

The improvement here is that helper doesn't need to filter all alerts (potentially a large number) just to always pick up the last element in the assertion.

Reply:
🤦 There was an unused function `GetLast AlertsWithDeploymentID` that does that. I've switch the code to use that one.